### DISCUSSION TITLE
要不要给组件加上 source map ？

### DISCUSSION THREAD

**Discussion by timongh** [the1812/Bilibili-Evolved#2795] [综合讨论 / General]
我看到 webpack/webpack.config.js 的 getDefaultConfig 函数里加过 source map 但是又注释掉了，是有什么考虑吗？

开这个讨论其实是我这两天在本地改了改配置，给加上了 source map，想提个 PR。但是加了 source map 后，可能需要将编译结果分成两个版本，debug 版本带 source map 且不上传，release 不带 source map，跟现在的 dist 功能一样。因此目录结构需要改。我想了两个方案

**方案一**
```
registry/
└── dist/
    ├── debug/
    │   ├── doc.js
    │   └── doc.js.map
    └── release/
        └── doc.js
```
这样的优点是结构比较清晰，缺点是现有文件结构需要改。我还没有研究修改结构在更新上是否有什么麻烦，有哪些地方需要同步修改。有知道的朋友说一下，或者我提个草稿 PR，大家帮着忙改。

**方案二**
```
registry/
└── dist/
    ├── components/
    ├── doc.js
    └── debug/
        ├── components/
        ├── doc.js
        └── doc.js.map
```
这样就不会影响原来的结构，但看起来没那么好看就是了。

大家有什么看法吗

**Comment by the1812** [comment]
默认的 development 模式下就带有内联的 source map 供调试使用, 放到单独文件的话能读取到吗?

**Reply by timongh** [reply]
放单独文件是可以的，改用插件并设置选项能做到这一点。

不过你的意思是，需要 source map 就将模式手动调成 development 吗？好吧，我之前确实没想到这么简单的方式，脑子锈了。那这个 PR 我就不提了吧。

**Reply by the1812** [reply]
也不用手动, core 和 registry 都配备了 dev / watch 类的任务, 这些任务生成的代码都是带 source map 的

我只是好奇放单独文件, 浏览器能读取得到吗, 代码自身都是用 `eval()` 执行的

**Reply by timongh** [reply]
改改配置能读。加了 source map 的代码最后都有一行注释 "//# sourceMappingURL=???"，这里只要改成 map 文件的本地路径就可以了。webpack.SourceMapDevToolPlugin 插件能够设置这些细节，文档在官网都有。

另外目前的 registry 默认不是没带 source map 吗。（core 没研究不清楚）

**Reply by the1812** [reply]
啊, registry 还真没加上, 那我改一下 webpack config 吧
